internal/order/order.go

```diff
@@ -7,14 +7,13 @@ package order
 import (
 	"cmp"
 	"slices"
-	"sort"
 
 	"gonum.org/v1/gonum/graph"
 )
 
 // ByID sorts a slice of graph.Node by ID.
-func ByID(n []graph.Node) {
-	sort.Slice(n, func(i, j int) bool { return n[i].ID() < n[j].ID() })
+func ByID[S interface{ ~[]E }, E graph.Node](n S) {
+	slices.SortFunc(n, func(a, b E) int { return cmp.Compare(a.ID(), b.ID()) })
 }
```

Comment:
I'm not sure this wins us anything.

Reply:
True, given the benchmark results and that we only use this function with `[]graph.Node`s and not subtypes. I'll revert the changes here.

By the way, how do you feel about the other `slices.SortFunc`s in this file diff, as well as the use of `min`?

Reply:
Oh, ignore my comment about using this function only with `[]graph.Node`s directly. Apparently the generics is needed to this function be used with `[]rdf.Term` and `[]graph.Line` slices in various areas of the codebase. So I'll just revert the `slices.SortFunc` part.

Reply:
Done.

My comment from earlier still stands:

> By the way, how do you feel about the other slices.SortFuncs in this file diff, as well as the use of min?

Reply:
The places where I haven't comments, I'm happy with. In some cases the use of that API is a clear clarity win.

Reply:
Sounds good to me! I'll set this conversation to resolved then.